src/tosyntax/FStar.ToSyntax.Resugar.fs

```diff
@@ -0,0 +1,614 @@
+    | "Colon" -> Some ":"
+    | _ -> None
+  in
+  match s with 
+  | "op_String_Assignment" -> Some ".[]<-"
+  | "op_Array_Assignment" -> Some ".()<-"
+  | "op_String_Access" -> Some ".[]"
+  | "op_Array_Access" -> Some ".()"
+  | Prefix "op__" op -> name_of_op op
+  | _ -> None
+
+let resugar_match_pat (p:S.pat) : A.pattern = 
+  (* We lose information when desugar PatAscribed to able to resugar it back *)
+  let mk a = A.mk_pattern a p.p in
+  let rec aux (p:S.pat) = 
+    match p.v with
+    | Pat_constant c -> mk (A.PatConst c)
+
+    | Pat_disj args ->
+      let args = List.map(fun p -> aux p) args in
+      mk (A.PatOr args)
+
+    | Pat_cons (fv, []) ->
+      mk (A.PatName fv.fv_name.v)
+
+    | Pat_cons(fv, args) when fv.fv_name.v.str.Equals C.cons_lid ->
+      failwith "PatList not resugared yet"
+
+    | Pat_cons(fv, args) when U.is_tuple_data_lid' fv.fv_name.v || U.is_dtuple_data_lid' fv.fv_name.v ->
+      failwith "PatTuple not resugared yet"
+
+    | Pat_cons({fv_qual=Some (Record_ctor _)}, args) ->
+      (* need to find the corresponding record and go through the record.fields *)
+      (* if the pattern is not Wild, then we have the pattern for the field to make the (lid*pattern) *)
+      failwith "PatRecord not resugared yet"
+
+    | Pat_cons (fv, args) ->
+      let args = List.map (fun (p, b) -> aux p) args in
+      mk (A.PatApp(mk (A.PatName fv.fv_name.v), args))
+
+    | Pat_var v -> 
+      (* both A.PatTvar and A.PatVar are desugared to S.Pat_var, only resugar back as PatVar*)
+      begin match string_to_op v.ppname.idText with
+       | Some op -> mk (A.PatOp op)
+       | None -> mk (A.PatVar (bv_as_unique_ident v, None))
+      end
+
+    | Pat_wild _ -> mk (A.PatWild)
+
+    | Pat_dot_term (bv, term) -> failwith "case Pat_dot_term is impossible"
+  in
+    aux p
+
+let resugar_term_as_op(t:S.term) : option<string> =
+  let infix_prim_ops = [
+    (C.op_Addition    , "+" );
+    (C.op_Subtraction , "-" );
+    (C.op_Minus       , "-" );
+    (C.op_Multiply    , "*" );
+    (C.op_Division    , "/" );
+    (C.op_Modulus     , "%" );
+    (C.read_lid       , "!" );
+    (C.list_append_lid, "@" );
+    (C.list_tot_append_lid,"@");
+    (C.strcat_lid     , "^" );
+    (C.pipe_right_lid , "|>");
+    (C.pipe_left_lid  , "<|");
+    (C.op_Eq          , "=" );
+    (C.op_ColonEq     , ":=");
+    (C.op_notEq       , "<>");
+    (C.not_lid        , "~" );
+    (C.op_And         , "&&");
+    (C.op_Or          , "||");
+    (C.op_LTE         , "<=");
+    (C.op_GTE         , ">=");
+    (C.op_LT          , "<" );
+    (C.op_GT          , ">" );
+    (C.op_Modulus     , "mod");
+    (C.and_lid     , "/\\");
+    (C.or_lid      , "\\/");
+    (C.imp_lid     , "==>");
+    (C.iff_lid     , "<==>");
+    (C.precedes_lid, "<<");
+    (C.eq2_lid     , "==");
+    (C.precedes_lid, "<<");
+    (C.eq3_lid     , "===");
+    (C.forall_lid  , "forall");
+    (C.exists_lid  , "exists");
+    (C.salloc_lid  , "alloc")
+  ] in
```

Comment:
It is very similar to the list in Fstar.Syntax.Print.fs, but included a few more operations. I can create a list just for the extra operations I need and then use the list Fstar.Syntax.Print.fs for the rest, would that be more acceptable?